src/Slider/Slider.jsx

```diff
@@ -403,9 +406,22 @@ const Slider = class Slider extends React.Component {
     });
   }
 
+
   play() {
-    const { playDirection } = this.props;
-    this.interval = setInterval(playDirection === 'forward' ? this.playForward : this.playBackward, this.props.interval);
+    if (this.props.intervalList) {
+      let priorIntervals = 0;
+      for (let i = 0; i < this.props.intervalList.length - 1; i += 1) {
+        setTimeout(this.props.playDirection === 'forward' ? this.playForward : this.playBackward, this.props.intervalList[i] + priorIntervals);
+        priorIntervals += this.props.intervalList[i];
+      }
+      setTimeout(this.props.playDirection === 'forward' ? this.playForward : this.playBackward,
+        this.props.intervalList[this.props.intervalList.length - 1] + priorIntervals);
+      setTimeout(this.play,
+        this.props.intervalList[this.props.intervalList.length - 1] + priorIntervals);
```

Comment:
I have looked at this block for a long time and I am still not 100% sure how it's working, When you have a moment would you be able to walk me through this ?

Reply:
I just pushed a commit that comments that area of code. I think a solution would be when a user skips to a different slide then we would reinitate the play function. Thoughts on this?